### DISCUSSION TITLE
`niri/workspaces` module for waybar

### DISCUSSION THREAD

**Discussion by egolep** [YaLTeR/niri#435] [Ideas]
I don't know if this is actually a feature request or if this already exists and I'm not finding it.
It would be useful to have something to show Niri's workspaces in Waybar like Sway and other WMs are doing.

**Comment by YaLTeR** [comment]
We need the event stream IPC first, then this would be a Waybar feature request I guess.

**Comment by lpnh** [comment]
In the meantime, the community is playing with some scripts to achieve this. Take a look at [this thread](https://github.com/YaLTeR/niri/discussions/356), maybe it can be helpful.

**Comment by magthe** [comment]
Wouldn't implementing support for `ext_workspace_unstable_v1` ([workspaces](https://github.com/Alexays/Waybar/wiki/Module:-Workspaces)) be the ideal option?

**Reply by YaLTeR** [reply]
Well kinda, we even have an issue #135, but last time I checked the protocol was in a bit of a limbo.

**Comment by YaLTeR** [comment]
I finished the event stream IPC in niri, so you can use a git build of niri with my Waybar PR: https://github.com/Alexays/Waybar/pull/3551
